ui/tests/integration/components/sync/secrets/page/overview-test.js

```diff
@@ -53,135 +53,263 @@ module('Integration | Component | sync | Page::Overview', function (hooks) {
+      await this.renderComponent();
+
+      assert.dom(overview.optInBanner).doesNotExist();
     });
+  });
 
-    await this.renderComponent();
+  module('with no destinations', function (hooks) {
+    hooks.beforeEach(function () {
+      this.destinations = [];
+    });
 
-    assert.dom(emptyStateTitle).hasText('Error fetching information', 'Empty state title renders');
-    assert
-      .dom(emptyStateMessage)
-      .hasText('Ensure that the policy has access to read sync associations.', 'Empty state message renders');
+    test('it should show a CTA', async function (assert) {
+      await this.renderComponent();
+
+      assert.dom(cta.button).exists();
+
+      assert
+        .dom(overview.createDestination)
+        .doesNotExist('create new destination link is hidden if there are no pre-existing destinations');
+    });
+
+    test('it should hide the overview cards', async function (assert) {
+      await this.renderComponent();
+
+      assert
+        .dom(overviewCard.title('Secrets by destination'))
+        .doesNotExist('it does not show secrets by destination card if there are no destinations');
+      assert
+        .dom(overviewCard.title('Total destinations'))
+        .doesNotExist('it does not show total destinations card if there are no destinations');
+      assert
+        .dom(overviewCard.title('Total secrets'))
+        .doesNotExist('it does not show total secrets card if there are no destinations');
+    });
   });
 
-  test('it should render totals cards', async function (assert) {
-    await this.renderComponent();
+  module('with destinations', function () {
+    test('it should show the overview cards', async function (assert) {
+      await this.renderComponent();
 
-    const { title, description, action, content } = overviewCard;
-    const cardData = [
-      {
-        cardTitle: 'Total destinations',
-        subText: 'The total number of connected destinations',
-        actionText: 'Create new',
-        count: '6',
-      },
-      {
-        cardTitle: 'Total secrets',
-        subText:
-          'The total number of secrets that have been synced from Vault. One secret will be counted as one sync client.',
-        actionText: 'View billing',
-        count: '7',
-      },
-    ];
-
-    cardData.forEach(({ cardTitle, subText, actionText, count }) => {
-      assert.dom(title(cardTitle)).hasText(cardTitle, 'Overview card title renders');
-      assert.dom(description(cardTitle)).hasText(subText, 'Destinations overview card description renders');
-      assert.dom(content(cardTitle)).hasText(count, 'Total count renders');
-      assert.dom(action(cardTitle)).hasText(actionText, 'Card action renders');
+      assert.dom(overviewCard.title('Secrets by destination')).exists();
+      assert.dom(overviewCard.title('Total destinations')).exists();
+      assert.dom(overviewCard.title('Total secrets')).exists();
+    });
+
+    module('Secrets by destination table', function () {
```

Comment:
the complexity of this module is a signal that this chunk of the component should be split into its own component. not gonna do that here, just thinking out loud 🙃